### PR TITLE
Configure SwiftLint via a YAML file

### LINKED ISSUE #1: Unable to specify what rules to validate against
We should be able to specify what to use - or perhaps specify what rules NOT to use. We should also probably be able to get a list of current rules.

### PR DESCRIPTION
**Pull request by jpsim**
Fixes #1 and #3.

### CODE REVIEW

**Comment by jpsim** [comment]
@keith I assume this would be useful for you, I'd love if you could share your thoughts on the implementation.

File: Source/SwiftLintFramework/Configuration.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                fputs("config error: '\(duplicateRule.0)' is listed \(duplicateRule.1) times\n",
+                    stderr)
+            }
+            return nil
+        }
+    }
+
+    public init?(yaml: String) {
+        guard let yamlConfig = Yaml.load(yaml).value else {
+            return nil
```
Comment (keith) [inline Source/SwiftLintFramework/Configuration.swift]: Should this print some error about an invalid config?

Comment (keith) [inline Source/SwiftLintFramework/Configuration.swift]: Ah I see this error below

File: Source/swiftlint/LintCommand.swift [hunk 2]
```diff
@@ -21,36 +21,43 @@ struct LintCommand: CommandType {
...
             if options.useSTDIN {
                 let standardInput = NSFileHandle.fileHandleWithStandardInput()
                 let stdinData = standardInput.readDataToEndOfFile()
                 let stdinNSString = NSString(data: stdinData, encoding: NSUTF8StringEncoding)
                 if let stdinString = stdinNSString as? String {
-                    let violations = Linter(file: File(contents: stdinString)).styleViolations
-                    print("\n".join(violations.map { $0.description }))
+                    let file = File(contents: stdinString)
+                    let linter = Linter(file: file, configuration: configuration)
+                    print("\n".join(linter.styleViolations.map { $0.description }))
```
Comment (keith) [inline Source/swiftlint/LintCommand.swift]: Not sure which beta you're on but this `.join` will have to change eventually

**Comment by keith** [comment]
Awesome! This looks great! :+1:

File: Source/swiftlint/LintCommand.swift (hunk 2 above)
Comment (jpsim) [inline Source/swiftlint/LintCommand.swift]: The `swift-2.0` branch is targeting Xcode 7 beta 5. I'll have to update all the dependencies to beta 6 before I can do the same to SwiftLint... ugh.

**Pull request merged**